pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -178,15 +160,9 @@ pub async fn handle_walreceiver_connection(
                     caught_up = true;
                 }
 
-                let timeline_to_check = Arc::clone(&timeline);
-                tokio::task::spawn_blocking(move || timeline_to_check.check_checkpoint_distance())
-                    .await
-                    .with_context(|| {
-                        format!("Spawned checkpoint check task panicked for timeline {id}")
-                    })?
-                    .with_context(|| {
-                        format!("Failed to check checkpoint distance for timeline {id}")
-                    })?;
+                timeline.check_checkpoint_distance().with_context(|| {
```

Comment:
That is still a blocking operation, isn't it? Not sure I understand why should we revert this.